picard/debug_opts.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+#
+# Picard, the next-generation MusicBrainz tagger
+#
+# Copyright (C) 2024 Laurent Monin
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+from collections import namedtuple
+from enum import IntEnum
+
+
+class DebugOpt(IntEnum):
+    WS_REPLIES = 1
+
+
+DebugOptDesc = namedtuple('DebugOptDesc', ('name', 'description'))
+
+debug_opts_descriptions = {
+    DebugOpt.WS_REPLIES: DebugOptDesc(name='ws_replies', description=N_('Log web service replies'))
+}
+
+
+debug_opts = set()
+
+
+def debug_opts_names():
+    """Returns a sorted list of all valid names for debug options"""
+    return sorted(v.name for v in debug_opts_descriptions.values())
+
+
+def debug_opts_help():
+    """Generates lines for a text describing each debug option"""
+    for o, d in debug_opts_descriptions.items():
+        yield '%20s: %s' % (d.name, _(d.description))
+
+
+def parse_command_line_debug_opts(string):
+    """Parse command line argument, a string with comma-separated values,
+    and enable corresponding debug options"""
+    opts = {str(o).strip().lower() for o in string.split(',')}
+    for debug_opt, opt_desc in debug_opts_descriptions.items():
+        if opt_desc.name in opts:
+            debug_opt_enable(debug_opt)
+
+
+def is_debug_opt_enabled(debug_opt):
+    """Returns True if passed debug option is enabled"""
+    return debug_opt in debug_opts
+
+
+def debug_opt_set(debug_opt, state):
+    """Set the boolean state of each debug option, stored in debug_opts"""
+    if state:
+        debug_opts.add(debug_opt)
+    else:
+        debug_opts.remove(debug_opt)
```

Comment:
I really like this new approach, but under what cirrcumstance do you expect this (remove) to occur?  Or is this included to make the functionality complete to handle future changes when the command line could be used to disable options that are normally set by default?

Reply:
When we toggle those by UI.

Reply:
Okay, I think I'll have to wait and see the code to fully understand.  Are you saying that omitting an item on the command line will toggle the option off if it is otherwise set (via the UI and a persistent setting)?

Reply:
I just looked at it again, and it all now makes sense to me.  I have no idea what I was thinking about before.  Sorry for the confusion.  (And I still really like this approach.)